### DISCUSSION TITLE
How do I prevent createContext from being called when creating a client on the server?

### DISCUSSION THREAD

**Discussion by SleeplessOne1917** [urql-graphql/urql#2829] [Q&A] [answered]
I have the following urql import:
```typescript
import { TypedDocumentNode, createClient } from 'urql/core';
```

I use the import in the following way:
```typescript
const client = createClient({
	url: `/api/graphql`
});
```

When I do this and try to use the client in a server component, I get an error that says `TypeError: react.createContext is not a function`. From looking around in the codebase, `urql/core` exports `@urql/core` which should create a client without even loading the file that has `createContext` in it. React context only gets used in the react and preact libraries, or at least so I thought.  How do I avoid that `createContext` code?

**Comment by SleeplessOne1917** [comment]
This is what I got on the Nextjs error page:
```
Server Error
TypeError: react.createContext is not a function

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Call Stack
eval
(sc_server)/node_modules/.pnpm/urql@3.0.3_onqnqwb3ubg5opvemcqf7c2qhy/node_modules/urql/dist/urql.js (15:20)
(sc_server)/./node_modules/.pnpm/urql@3.0.3_onqnqwb3ubg5opvemcqf7c2qhy/node_modules/urql/dist/urql.js
file:///home/abias/projects/5E-Character-Sheet-Builder/.next/server/app/create/page.js (3710:1)
__webpack_require__
file:///home/abias/projects/5E-Character-Sheet-Builder/.next/server/webpack-runtime.js (33:43)
eval
webpack-internal:///(sc_server)/./src/graphql/queries/CharacterSheetBuilder/spells/getSpell.ts (5:62)
(sc_server)/./src/graphql/queries/CharacterSheetBuilder/spells/getSpell.ts
file:///home/abias/projects/5E-Character-Sheet-Builder/.next/server/app/create/page.js (3775:1)
__webpack_require__
file:///home/abias/projects/5E-Character-Sheet-Builder/.next/server/webpack-runtime.js (33:43)
eval
webpack-internal:///(sc_server)/./src/graphql/characterSheetBuilderClientService.ts (10:104)
(sc_server)/./src/graphql/characterSheetBuilderClientService.ts
file:///home/abias/projects/5E-Character-Sheet-Builder/.next/server/app/create/page.js (3753:1)
__webpack_require__
file:///home/abias/projects/5E-Character-Sheet-Builder/.next/server/webpack-runtime.js (33:43)
eval
webpack-internal:///(sc_server)/./app/create/page.tsx (8:105)
```

**Reply by SleeplessOne1917** [reply]
characterSheetBuilderClientService.ts (10:104) is where I tried defining my client.

**Comment by JoviDeCroock** [comment, marked as answer]
Server-Components do not support `createContext` hence if you want to use urql in an import of a server-component or the component itself you will need to use `@urql/core` as described [here](https://github.com/urql-graphql/urql/issues/2799) and [here](https://github.com/urql-graphql/urql/discussions/2800). The issue also links to the relevant Next.JS issue

**Reply by SleeplessOne1917** [reply]
I was already using `@urql/core`. The problem ended up being that while I was importing `createClient` from `@urql/core`, for the queries I was importing `gql` from `urql`. That ended up being the issue.
